### PR TITLE
Run Unmarked Test pre-commit only if all other hooks pass

### PR DESCRIPTION
**Pull request by dpanici**
the unmarked test pre-commit hook (at least on my laptop) can take anywhere from 5-10 seconds to run, so I'd rather it only run if everything else is passing.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ repos:
     rev: 24.3.0
     hooks:
       - id: black-jupyter
+        log_file: devtools/pre-commit.log
```
Comment (dpanici) [inline .pre-commit-config.yaml:6]: this log is only created if black fails

File: devtools/clean_logfiles.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+if [ -f "devtools/pre-commit.log" ]; then
+   echo "delete devtools/pre-commit.log"
```
Comment (YigitElma) [inline devtools/clean_logfiles.sh:4]:
```suggestion
   echo "deleting devtools/pre-commit.log file"
```

**Review by YigitElma — APPROVED** [review]

**Review by YigitElma — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (f0uriest) [reply .pre-commit-config.yaml:6]: if it fails on the first pass, the run again does it delete the old log file? or does it just overwrite it with a blank one?

Comment (YigitElma) [reply .pre-commit-config.yaml:6]: I think the last one deletes it at the end of each run.

**Review by f0uriest — APPROVED** [review]

**Pull request merged**